data_collection/gazette/spiders/sp_paulinia.py

```diff
@@ -0,0 +1,81 @@
+import datetime
+
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class SpPauliniaSpider(BaseGazetteSpider):
+    name = "sp_paulinia"
+    TERRITORY_ID = "2905206"
+    start_date = datetime.date(2012, 1, 4)
+    allowed_domains = ["www.paulinia.sp.gov.br"]
+    start_urls = ["http://www.paulinia.sp.gov.br/semanarios"]
+
+    """
+    No site de Paulínia, os diários são disponibilizados em páginas por ano, agrupadas em colunas
+    de meses, onde cada edição de diário oficial está listado. Não há filtros por data. 
+
+    Por isso, a lógica de raspagem segue a 'navegação humana': primeiro acessa a página principal, 
+    a partir dela, se obtém os diários ao escolher o ano, e percorrer os meses e as edições listadas.
+    """
+
+    def parse(self, response):
+        # coleta a lista de anos disponíveis
+        years = response.css("div.col-md-1")
+
+        # percorre cada um dos anos disponíveis
+        for year in years:
+            year_to_scrape = int(year.xpath("./a/text()").get())
+
+            # verifica se o ano sendo tratado está no intervalo da raspagem
+            if not (self.start_date.year <= year_to_scrape <= self.end_date.year):
+                continue  # pula a iteração atual caso não esteja
+
+            # descobre qual o código do evento que acontece no site quando se "clica" no ano
+            event_target = year.xpath("./a/@href").re_first(r"(ctl00.*?)',")
+
+            # solicita uma requisição usando o código do evento para aquele ano
+            yield scrapy.FormRequest.from_response(
+                response,
+                formdata={"__EVENTTARGET": event_target},
+                callback=self.parse_year,
+            )
+
+        # chama a função parse_year() e devolve o mesmo conteúdo que ela devolver
+        yield from self.parse_year(response)
+
+    def parse_year(self, response):
+        # a partir da resposta da página, usa o xpath para obter todas as referências aos diários
+        editions = response.xpath(
+            "//div[@class='container body-content']//div[@class='row']//a[contains(@href, 'AbreSemanario')]"
+        )
```

Comment:
Podemos substituir por seletor CSS e vai ficar melhor de lidar com as classes
```suggestion
        editions = response.css("div.body-content div.row a[href*='AbreSemanario']")
```